<launch>
<arg name="remapped_tf" default="/tf" />
<arg name="output" default="log" />
<arg name="input_point_cloud" default="/rgbd_longrange/depth/points_xyzrgb" />
<arg name="output_point_cloud" default="/octomap_input" />
<arg name="relay_topic_prefix" default="" />
<arg name="joint_topic_prefix" default="/"/>


 
<param name="left_gripper_description" textfile="$(find hose_localization)/gripper_descriptions/left_gripper_v3.urdf"/>
<param name="right_gripper_description" textfile="$(find hose_localization)/gripper_descriptions/right_gripper_v3.urdf"/>

       <node name="left_gripper_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher">
       	     <remap from="robot_description" to="left_gripper_description" />
	     <remap from="tf" to="$(arg remapped_tf)" />
	     <remap from="joint_states" to="$(arg relay_topic_prefix)$(arg joint_topic_prefix)joint_states" />
             <param name="tf_prefix" value="left_gripper"/>
       </node>	    

 <node name="right_gripper_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher">
       	     <remap from="robot_description" to="right_gripper_description" />
             <param name="tf_prefix" value="right_gripper"/>
     	     <remap from="tf" to="$(arg remapped_tf)" />	
	     <remap from="joint_states" to="$(arg relay_topic_prefix)$(arg joint_topic_prefix)joint_states" />
       </node>	    

    <node name="left_gripper_marker" pkg="hose_localization" type="gripper_marker.py" output="$(arg output)" >
      <remap from="/tf" to="$(arg remapped_tf)" />
      <!--<param name="~user_ns_" value="/left_gripper_marker"/> -->
      <remap from="joint_states" to="$(arg relay_topic_prefix)/joint_states" />
    </node> 


    <node name="right_gripper_marker" pkg="hose_localization" type="gripper_marker.py" output="screen" >
      <remap from="/tf" to="$(arg remapped_tf)" />
      <param name="marker_namespace" value="/right_gripper_marker"/>
      <remap from="joint_states" to="$(arg relay_topic_prefix)/joint_states" />
    </node> 



    <node name="graspable_cylinder" pkg="hose_localization" type="graspable_cylinder_marker.py" output="screen" >
        <remap from="/tf" to="$(arg remapped_tf)" /> 			
        <remap from="/snapshot_input_cloud" to="$(arg input_point_cloud)" />
	<remap from="/output_cloud" to="$(arg output_point_cloud)"/>
        
    </node> 
    <node name="wye_marker" pkg="hose_localization" type="wye_marker.py" output="screen" >

        <remap from ="/tf" to="$(arg remapped_tf)" /> 			
        
    </node> 

    <node name="reel_marker" pkg="hose_localization" type="reel_marker.py" output="screen" >
        <remap from="/tf" to="$(arg remapped_tf)" /> 			
    </node> 


</launch>
